### PR TITLE
feat(NcRichText): make interactive widgets opt-in

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

- Do not load widget integration by default (if embedded e.g. in Talk)
- Fix min-height of list (test 'before' -> when loading is done, list shrinks)
- Make toggle button present (remove opacity, move to corner)

### 🖼️ Screenshots

🏚️ Before

https://github.com/user-attachments/assets/15384f8a-9f78-4770-adc4-75d256149806

🏡 After

https://github.com/user-attachments/assets/14f260ac-7538-49d3-9d14-efe554d1a832

### 🏁 Checklist

- [ ] ⛑️ Tests are included or are not applicable
- [ ] 📘 Component documentation has been extended, updated or is not applicable
- [ ] 3️⃣ Backport to `next` requested with a Vue 3 upgrade

### CODE REVIEW

File: src/components/NcRichText/NcReferenceWidget.vue [hunk 1]
```diff
@@ -317,17 +317,9 @@ export default {
 	position: relative;
 	.toggle-interactive--button {
 		position: absolute;
-		top: 50%;
+		bottom: var(--default-grid-baseline);
+		right: var(--default-grid-baseline);
```
Comment (ShGKme) [inline src/components/NcRichText/NcReferenceWidget.vue:321]: IMO, it looks better, but it can overflow the content

**Review by ShGKme — APPROVED** [review]

File: src/components/NcRichText/NcReferenceWidget.vue (hunk 1 above)
Comment (Antreesy) [reply src/components/NcRichText/NcReferenceWidget.vue:321]: We can consider:
- make a button semi-transparent when not hovered;
- make it icon button (and put text in title/aria-label

Don't see it moved outside of widget area

File: src/components/NcRichText/NcReferenceList.vue [hunk 2]
```diff
@@ -146,11 +148,8 @@ export default {
 }
 </script>
 <style lang="scss" scoped>
-.widget--list {
+.widgets--list {
 	width: var(--widget-full-width, 100%);
```
Comment (juliusknorr) [inline src/components/NcRichText/NcReferenceList.vue:152]: This breaks text styling as we set a custom full width to span the whole page, however the actual styling is applied in the NcReferenceWidget component.

This should fix it:

```suggestion
.widgets--list {
	width: 100%;
```

Comment (Antreesy) [reply src/components/NcRichText/NcReferenceList.vue]: yup, wasn't noticed before of wrong classname, probably?

Comment (juliusknorr) [reply src/components/NcRichText/NcReferenceList.vue]: Yes, I guess so

**Review by juliusknorr — APPROVED** [review]

**Review by DorraJaouad — APPROVED** [review]
👍🏻 👍🏻

**Pull request merged**